src/interfaces/User.ts

```diff
@@ -0,0 +1,14 @@
+export interface ICar {
+  brand: string;
+  color: string;
+}
+
+export interface IUserContext {
+  name: string;
+  lastName: string;
+  isLoggedIn: boolean;
+  cars: ICar[];
+  addCar?: () => void;
```

Comment:
here you are mixing your functions with the initial state, I'd suggest to separate this to have an interface only for the state